worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/Flusher.scala

```diff
@@ -110,15 +107,23 @@ abstract private[worker] class Flusher(
     buffer
   }
 
-  def returnBuffer(buffer: CompositeByteBuf): Unit = {
-    MemoryManager.instance().releaseDiskBuffer(buffer.readableBytes())
+  def returnBuffer(task: FlushTask): Unit = {
+    val bufferSize = task.buffer.readableBytes()
+    MemoryManager.instance().releaseDiskBuffer(bufferSize)
     Option(CongestionController.instance())
       .foreach(
-        _.consumeBytes(buffer.readableBytes()))
+        _.consumeBytes(bufferSize))
+    returnBuffer(task.buffer, task.keepBuffer)
+  }
+
+  def returnBuffer(buffer: CompositeByteBuf, keepBuffer: Boolean = false): Unit = {
     buffer.removeComponents(0, buffer.numComponents())
     buffer.clear()
-
-    bufferQueue.put(buffer)
+    if (keepBuffer) {
+      bufferQueue.put(buffer)
+    } else {
+      buffer.release()
```

Comment:
`buffer.removeComponents(0, buffer.numComponents())` already releases the  buffer:
```
    for (int i = cIndex; i < endIndex; ++i) {
            Component c = components[i];
            if (c.length() > 0) {
                needsUpdate = true;
            }
            if (lastAccessed == c) {
                lastAccessed = null;
            }
            c.free();
        }
```
So why call `buffer.release()` again?

Reply:
Also, since `buffer.removeComponents(0, buffer.numComponents())` removes all its components, IMO we can always keep the buffer?